Plugins/Player/Player.cpp

```diff
@@ -1249,11 +1249,21 @@ ArgumentStack Player::PossessCreature(ArgumentStack&& args)
                             pPossessor->RemoveAssociate(possessedOidPOS);
                             pPOS->Remove(pPossessor->m_idSelf, "possessedOid");
                             pPOS->Remove(possessedOidPOS, "possessorOid");
+
+                            auto possessedAssociateType = *pPOS->Get<int>(pPossessor->m_idSelf, "possessedAssociateType");
```

Comment:
From https://en.cppreference.com/w/cpp/utility/optional/operator* :
> The behavior is undefined if `*this` does not contain a value. 

I guess it just default-constructed it, but better to clear it up. I'd write this as:

```C
auto possessedAssociateType = pPOS->Get<int>(...);
if (possessedAssociateType && *possessedAssociateType != 0)
{
    pPossessor->AddAssociate(possessedOidPOS, *possessedAssociateType);
    ...
}
```

Reply:
You can also do:
```C
auto possessedAssociateType = pPOS->Get<int>(...).value_or(0);
```

Reply:
Thanks for the review. I'll update the PR tomorrow.